### PR TITLE
Learn CodeQL docs: add short note about new terminology

### PR DESCRIPTION
**Pull request by jf205**
Note explaining the use of CodeQL, QL and CodeQL databases in the Learn CodeQL Sphinx project.

### CODE REVIEW

File: docs/language/learn-ql/terminology-note.rst [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+The name QL now only refers to the query language that powers CodeQL analysis.
+
+The CodeQL queries and libraries used to analyze source code are written in QL.
+These queries and libraries are open source, and can be found in the `CodeQL repository <https://github.com/semmle/ql>`__.
+QL is a general-purpose, object-oriented language that can be used to query any kind of data. 
+
+CodeQL databases
+----------------
+
+QL snapshots have been renamed QL snapshots. CodeQL databases contain relational data created and analyzed using CodeQL. They are the equivalent of QL snapshots, but have been optimized for use with the CodeQL tools.
```
Comment (max-schaefer) [inline docs/language/learn-ql/terminology-note.rst]: > QL snapshots have been renamed QL snapshots.

:question:

File: docs/language/learn-ql/index.rst [hunk 2]
```diff
@@ -9,6 +9,18 @@ Here, you can query open source projects directly, without having to download Co
 
 CodeQL is based on a powerful query language called QL. The following topics help you understand QL in general, as well as how to use it when analyzing code with CodeQL.
 
+.. pull-quote::
+
+   Important 
+
+   If you've previously used QL, you may notice some slight changes in terms we use to describe some important concepts. For more information, see our note about :doc:`Recent terminology changes <terminology-note>`.
```
Comment (shati-patel) [inline docs/language/learn-ql/index.rst]:
```suggestion
   If you've previously used QL, you may notice slight changes in terms we use to describe some important concepts. For more information, see our note about :doc:`Recent terminology changes <terminology-note>`.
```
(Remove one of the "some"s)

File: docs/language/learn-ql/terminology-note.rst (hunk 1 above)
Comment (jf205) [reply docs/language/learn-ql/terminology-note.rst]: 🤦‍♂

File: docs/language/learn-ql/terminology-note.rst [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+The name QL now only refers to the query language that powers CodeQL analysis.
+
+The CodeQL queries and libraries used to analyze source code are written in QL.
+These queries and libraries are open source, and can be found in the `CodeQL repository <https://github.com/semmle/ql>`__.
+QL is a general-purpose, object-oriented language that can be used to query any kind of data. 
+
+CodeQL databases
+----------------
+
+QL snapshots have been renamed CodeQL databases. CodeQL databases contain relational data created and analyzed using CodeQL. They are the equivalent of QL snapshots, but have been optimized for use with the CodeQL tools.
```
Comment (shati-patel) [inline docs/language/learn-ql/terminology-note.rst]:
```suggestion
QL snapshots have been renamed CodeQL databases. :doc:`CodeQL databases <database>` contain relational data created and analyzed using CodeQL. They are the equivalent of QL snapshots, but have been optimized for use with the CodeQL tools.
```

Perhaps it's worth linking to the general topic about databases?

File: docs/language/learn-ql/terminology-note.rst [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+The name QL now only refers to the query language that powers CodeQL analysis.
+
+The CodeQL queries and libraries used to analyze source code are written in QL.
+These queries and libraries are open source, and can be found in the `CodeQL repository <https://github.com/semmle/ql>`__.
+QL is a general-purpose, object-oriented language that can be used to query any kind of data. 
+
+CodeQL databases
+----------------
+
+QL snapshots have been renamed CodeQL databases. :doc:`CodeQL databases <database>` contain relational data created and analyzed using CodeQL. They are the equivalent of QL snapshots, but have been optimized for use with the CodeQL tools.
```
Comment (felicitymay) [inline docs/language/learn-ql/terminology-note.rst]: Will this link work outside the CodeQL project?

**Comment by jf205** [comment]
Thanks for the review @felicitymay. 

> Also, I suggest that we update the home page question "What is CodeQL?" to link to this topic for more information.

I'm not sure which question you mean?

File: docs/language/learn-ql/terminology-note.rst (hunk 4 above)
Comment (felicitymay) [reply docs/language/learn-ql/terminology-note.rst]: Ah - sorry, assumed that it was for the glossary in the CodeQL documentation!

**Comment by felicitymay** [comment]
The question towards the bottom of the page: http://docteam.internal.semmle.com/felicity/help.semmle.com-SD-4051/help/home.html

**Comment by felicitymay** [comment]
@shati-patel - do you have any more feedback or is this okay to merge?

**Pull request merged**